share/util/shr_flux_mod.F90

```diff
@@ -362,13 +368,31 @@ SUBROUTINE shr_flux_atmOcn(nMax  ,zbot  ,ubot  ,vbot  ,thbot ,   &
         !------------------------------------------------------------
         !--- neutral coefficients, z/L = 0.0 ---
         stable = 0.5_R8 + sign(0.5_R8 , delt)
-        rdn    = sqrt(cdn(vmag))
+        if (wav_atm_coup .eq. 'twoway') then
+           if (z0wav(n) == 0.0_R8 ) then 
+              cdn_wav = cdn_wave(loc_karman,zref,0.0001_R8)
+           else
+              cdn_wav = cdn_wave(loc_karman,zref,z0wav(n))
+           endif
+           rdn = sqrt(cdn_wav)
+           ! rdn calculated from Z0 will be constant
+        else
+          rdn    = sqrt(cdn(vmag))
+        endif
         rhn    = (1.0_R8-stable) * 0.0327_R8 + stable * 0.018_R8
                  !(1.0_R8-stable) * chxcdu + stable * chxcds
         ren    = 0.0346_R8 !cexcd
 
         !--- ustar, tstar, qstar ---
-        ustar = rdn * vmag
+        if (wav_atm_coup .eq. 'twoway') then
+           if (ustarwav(n) == 0.0_R8 ) then 
+              ustar = ustarwav(n)+0.1_R8
```

Comment:
I'm not sure what this is doing.  if `ustarwav(n) == 0` why add it again?  Why add a hard coded 0.1?